client/src/app/components/gui-editor/gui-editor.component.ts

```diff
@@ -0,0 +1,93 @@
+import {
+    Component,
+    Input,
+    trigger,
+    style,
+    animate,
+    state,
+    transition
+} from "@angular/core";
+import {NgFor} from "@angular/common";
+import {FileModel} from "../../store/models/fs.models";
+import {PropertyInputComponent} from "../forms/inputs/property-input.component";
+import {GuiEditorService} from "./gui-editor.service";
+import {EditorSidebarComponent} from "./sidebar/editor-sidebar.component";
+import {PropertyPosition, VisibilityState} from "./animation.states";
+import {CommandLineComponent} from "./commandline/commandline.component";
+
+require("./gui-editor.component.scss");
+
+@Component({
+    selector: "gui-editor",
+    providers: [GuiEditorService],
+    directives: [NgFor, PropertyInputComponent, EditorSidebarComponent, CommandLineComponent],
```

Comment:
You don't need to import core directives like `NgFor` anymore: http://stackoverflow.com/questions/34385566/how-to-import-ngfor-in-angular-2-using-javascript#answer-34391950